src/engine.ml

```diff
@@ -288,29 +288,32 @@ let prf ?sids ~label ~secret ~client_random ~server_random len =
   and sha = p_hash Mirage_crypto.Hash.SHA1.(hmac, digest_size) s2 in
   Mirage_crypto.Uncommon.Cs.xor md5 sha
 
-let derive_keys session (own_key_material : State.own_key_material)
-    (peer_key_material : Packet.tls_data) =
-  let ( pre_master,
-        client_random,
-        server_random,
-        client_random',
-        server_random',
-        sids ) =
-    if Cstruct.(equal empty own_key_material.pre_master) then
+let derive_keys session (my_key_material : State.my_key_material)
+    (their_key_material : Packet.tls_data) =
+  let ( server,
+        ( pre_master,
+          client_random,
+          server_random,
+          client_random',
+          server_random',
+          sids ) ) =
+    if Cstruct.(equal empty my_key_material.pre_master) then
```

Comment:
Maybe bind `server` outside:
```suggestion
  let server = Cstruct.is_empty my_key_material.pre_master in
  let ( pre_master,
        client_random,
        server_random,
        client_random',
        server_random',
        sids ) =
    if server then
```

Reply:
The two branches of the `if`-expression needs to be altered accordingly, too. The GitHub interface is a bit painful for this...